qiita_db/base.py

```diff
@@ -128,28 +130,36 @@ def _check_id(self, id_):
         the other classes. However, still defining here as there is only one
         subclass that doesn't follow this convention and it can override this.
         """
-        self._check_subclass()
-
-        conn_handler = SQLConnectionHandler()
-
-        return conn_handler.execute_fetchone(
-            "SELECT EXISTS(SELECT * FROM qiita.{0} WHERE "
-            "{0}_id=%s)".format(self._table), (id_, ))[0]
-
-    def __init__(self, id_):
+        sql = "SELECT EXISTS(SELECT * FROM qiita.{0} WHERE {0}_id=%s)".format(
+            self._table)
+        trans.add(sql, [id_])
+        # The value that we want is the result of the last SQL query,
+        # and it is stored in the first value of the first row
+        return trans.execute()[-1][0][0]
```

Comment:
You can't centralize the indexing of [-1][0][0], because from the queue system you don't know if it returns a single value or multiple values. Unless with that parameter we document it really well and we say that it just does like [-1][0][0]... is that what you're proposing?